bundles/org.openhab.voice.whisperstt/src/main/java/org/openhab/voice/whisperstt/internal/WhisperSTTService.java

```diff
@@ -0,0 +1,656 @@
+        // Overwrite whisper jni shared library
+        Path whisperJNILinuxLibrary = libFolder.resolve("libwhisperjni.so");
+        Path whisperJNIMacLibrary = libFolder.resolve("libwhisperjni.dylib");
+        Path whisperJNIWinLibrary = libFolderWin.resolve("libwhisperjni.dll");
+        if (Files.exists(whisperJNILinuxLibrary)) {
+            options.whisperJNILib = whisperJNILinuxLibrary;
+        } else if (Files.exists(whisperJNIMacLibrary)) {
+            options.whisperJNILib = whisperJNIMacLibrary;
+        } else if (Files.exists(whisperJNIWinLibrary)) {
+            options.whisperJNILib = whisperJNIWinLibrary;
+        }
+        // Overwrite whisper shared library, Windows searches library in $env:PATH
+        Path whisperLinuxLibrary = libFolder.resolve("libwhisper.so");
+        Path whisperMacLibrary = libFolder.resolve("libwhisper.dylib");
+        if (Files.exists(whisperLinuxLibrary)) {
+            options.whisperLib = whisperLinuxLibrary;
+        } else if (Files.exists(whisperMacLibrary)) {
+            options.whisperLib = whisperMacLibrary;
+        }
+        // Log library registration
+        options.logger = (msg) -> logger.debug("Library load: {}", msg);
+        return options;
+    }
+
+    @Modified
+    protected void modified(Map<String, Object> config) {
+        configChange(config);
+    }
+
+    @Deactivate
+    protected void deactivate(Map<String, Object> config) {
+        try {
+            WhisperGrammar grammar = this.grammar;
+            if (grammar != null) {
+                grammar.close();
+                this.grammar = null;
+            }
+            unloadContext();
+        } catch (IOException e) {
+            logger.warn("IOException unloading model: {}", e.getMessage());
+        }
+        WhisperJNI.setLibraryLogger(null);
+    }
+
+    private void configChange(Map<String, Object> config) {
+        this.config = new Configuration(config).as(WhisperSTTConfiguration.class);
+        WhisperJNI.setLibraryLogger(this.config.enableWhisperLog ? this::onWhisperLog : null);
+        WhisperGrammar grammar = this.grammar;
+        if (grammar != null) {
+            grammar.close();
+            this.grammar = null;
+        }
+        WhisperJNI whisper;
+        try {
+            whisper = getWhisper();
+        } catch (IOException ignored) {
+            logger.warn("library not loaded, the add-on will not work");
+            return;
+        }
+        String grammarText = String.join("\n", this.config.grammarLines);
+        if (this.config.useGrammar && isValidGrammar(grammarText)) {
+            try {
+                logger.debug("Parsing GBNF grammar...");
+                this.grammar = whisper.parseGrammar(grammarText);
+            } catch (IOException e) {
+                logger.warn("Error parsing grammar: {}", e.getMessage());
+            }
+        }
+        if (this.config.preloadModel) {
+            try {
+                var ignored = loadContext();
```

Comment:
To avoid a warning, just call
```
loadContext();
```
without assigning to a variable that is then not used.